### PR TITLE
Update device option descriptions to accurately convey compatibility with containers and VMs

### PR DESCRIPTION
**Pull request by kadinsayani**
Resolves https://github.com/canonical/lxd/issues/14451.

### CODE REVIEW

File: lxd/device/unix_hotplug.go [hunk 1]
```diff
@@ -72,7 +72,14 @@ func (d *unixHotplug) validateConfig(instConf instance.ConfigReader) error {
 		"uid":       unixValidUserID,
 		"gid":       unixValidUserID,
 		"mode":      unixValidOctalFileMode,
-		"required":  validate.Optional(validate.IsBool),
+
+		// lxdmeta:generate(entities=device-unix-hotplug; group=device-conf; key=required)
```
Comment (tomponline) [inline lxd/device/unix_hotplug.go:76]: this should be a separate commit please

File: doc/metadata.txt [hunk 2]
```diff
@@ -1588,15 +1588,15 @@ The default is `false`, which means that all devices can be hotplugged.
 ```{config:option} gid device-unix-usb-device-conf
 :condition: "container"
 :defaultdesc: "`0`"
-:shortdesc: "GID of the device owner in the container"
+:shortdesc: "GID of the device owner in the instance"
```
Comment (minaelee) [inline doc/metadata.txt:1591]: Two lines before the changed line says:
```:condition: "container"``` 
(line 1589)

Does this mean that this option only exists on containers? Can you confirm whether this device option is also available to instances, and if so, what the `condition` property refers to? If this is an option that is only available when using a container, then the shortdescs should stay as-is.

Same for the other two similar changes on lines 1599 and 1626.

File: lxd/device/unix_common.go [hunk 3]
```diff
@@ -63,87 +63,92 @@ func (d *unixCommon) validateConfig(instConf instance.ConfigReader) error {
...
 
-		// lxdmeta:generate(entities=device-unix-usb; group=device-conf; key=uid)
-		//
+		// lxdmeta:generate(entities=device-unix-usb; group=device-conf; key=uid)//
 		// ---
 		//  type: integer
 		//  defaultdesc: `0`
 		//  condition: container
-		//  shortdesc: UID of the device owner in the container
+		//  shortdesc: UID of the device owner in the instance
```
Comment (minaelee) [inline lxd/device/unix_common.go:103]: Again here, the comments say ```condition: container``` - does this UID exist for instances?

File: lxd/device/unix_common.go [hunk 4]
```diff
@@ -63,87 +63,92 @@ func (d *unixCommon) validateConfig(instConf instance.ConfigReader) error {
...
 
-		// lxdmeta:generate(entities=device-unix-usb; group=device-conf; key=gid)
-		//
+		// lxdmeta:generate(entities=device-unix-usb; group=device-conf; key=gid)//
 		// ---
 		//  type: integer
 		//  defaultdesc: `0`
 		//  condition: container
-		//  shortdesc: GID of the device owner in the container
+		//  shortdesc: GID of the device owner in the instance
```
Comment (minaelee) [inline lxd/device/unix_common.go:118]: Does this GID exist for instances?

File: lxd/device/unix_common.go [hunk 5]
```diff
@@ -63,87 +63,92 @@ func (d *unixCommon) validateConfig(instConf instance.ConfigReader) error {
...
+		//  shortdesc: Mode of the device in the container
 
 		// lxdmeta:generate(entities=device-unix-usb; group=device-conf; key=mode)
 		//
 		// ---
 		//  type: integer
 		//  defaultdesc: `0660`
 		//  condition: container
-		//  shortdesc: Mode of the device in the container
+		//  shortdesc: Mode of the device in the instance
```
Comment (minaelee) [inline lxd/device/unix_common.go:134]: Does this config exist for instances?

File: lxd/metadata/configuration.json [hunk 6]
```diff
@@ -1837,8 +1837,8 @@
 						"gid": {
 							"condition": "container",
 							"defaultdesc": "`0`",
-							"longdesc": "",
-							"shortdesc": "GID of the device owner in the container",
+							"longdesc": "//",
+							"shortdesc": "GID of the device owner in the instance",
```
Comment (minaelee) [inline lxd/metadata/configuration.json:1841]: Does GID exist for instances?

File: lxd/metadata/configuration.json [hunk 7]
```diff
@@ -1847,7 +1847,7 @@
 							"condition": "container",
 							"defaultdesc": "`0660`",
 							"longdesc": "",
-							"shortdesc": "Mode of the device in the container",
+							"shortdesc": "Mode of the device in the instance",
```
Comment (minaelee) [inline lxd/metadata/configuration.json:1850]: Does device mode exist for instances?

File: lxd/metadata/configuration.json [hunk 8]
```diff
@@ -1877,8 +1877,8 @@
 						"uid": {
 							"condition": "container",
 							"defaultdesc": "`0`",
-							"longdesc": "",
-							"shortdesc": "UID of the device owner in the container",
+							"longdesc": "//",
+							"shortdesc": "UID of the device owner in the instance",
```
Comment (minaelee) [inline lxd/metadata/configuration.json:1881]: Does device owner UID exist for instances?

**Review by minaelee — CHANGES_REQUESTED** [review]
Please confirm if the usb device `guid`, `uid`, and device `mode` config options exist for instances, or only for containers. If the former, then the changes from container -> instance should be made. If the latter, the changes are not needed.

Could you please also update the Markdown docs pages that pull in the changes in this PR so that they align with these changes? Example in [unix-char](https://documentation.ubuntu.com/lxd/en/latest/reference/devices_unix_char/):
>Unix character devices make the specified character device appear as a device in the instance (under /dev). 

Here, `instances` should be changed to `containers`.

The pages needing updates (re-linking here for convenience from #14451):
[unix-char](https://documentation.ubuntu.com/lxd/en/latest/reference/devices_unix_char/)
[unix-block](https://documentation.ubuntu.com/lxd/en/latest/reference/devices_unix_block/)
[usb](https://documentation.ubuntu.com/lxd/en/latest/reference/devices_usb/#)
[unix-hotplug](https://documentation.ubuntu.com/lxd/en/latest/reference/devices_unix_hotplug/)

Thanks! I appreciate the attention to consistent terminology, and so will the users :)

File: doc/metadata.txt (hunk 2 above)
Comment (kadinsayani) [reply doc/metadata.txt:1591]: The `usb` device type is available for both containers and VMs. The idea here is to consistently use the word `instance` for `usb` device options, but keep the `container` condition since the `uid`, `gid`, and `mode` options only apply to containers.

**Review by minaelee — APPROVED** [review]

File: lxd/metadata/configuration.json [hunk 9]
```diff
@@ -1877,8 +1877,8 @@
 						"uid": {
 							"condition": "container",
 							"defaultdesc": "`0`",
-							"longdesc": "",
-							"shortdesc": "UID of the device owner in the container",
+							"longdesc": "//",
```
Comment (tomponline) [inline lxd/metadata/configuration.json:1880]: Why "//"?

Comment (kadinsayani) [reply lxd/metadata/configuration.json]: Formatting issue. It's fixed now.

**Review by minaelee — APPROVED** [review]

**Review by tomponline — APPROVED** [review]
Thanks!
